apps/journeys-admin/src/components/Editor/Slider/JourneyFlow/libs/useCreateStep/useCreateStep.spec.tsx

```diff
@@ -0,0 +1,273 @@
+import { defaultJourney } from '../../../../data'
+
+import { useCreateStep } from '.'
+
+jest.mock('uuid', () => ({
+  __esModule: true,
+  v4: jest.fn()
+}))
+
+const mockUuidv4 = uuidv4 as jest.MockedFunction<typeof uuidv4>
+
+const step = {
+  __typename: 'StepBlock',
+  id: 'step.id'
+} as unknown as TreeBlock<StepBlock>
+
+blockOrderUpdateMock.request.variables = {
+  id: 'newStep.id',
+  journeyId: 'journey-id',
+  parentOrder: 0
+}
+blockOrderUpdateMock.result = jest.fn(() => ({
+  data: {
+    blockOrderUpdate: [
+      {
+        __typename: 'StepBlock',
+        id: 'blockId',
+        parentOrder: 0
+      }
+    ]
+  }
+}))
+
+navigateToBlockActionUpdateMock.request.variables = {
+  id: 'block1.id',
+  journeyId: 'journey-id',
+  input: {
+    blockId: 'newStep.id'
+  }
+}
+navigateToBlockActionUpdateMock.result = jest.fn(() => ({
+  data: {
+    blockUpdateNavigateToBlockAction: {
+      __typename: 'NavigateToBlockAction',
+      parentBlockId: 'step1.id',
+      gtmEventName: null,
+      blockId: 'newStep.id'
+    }
+  }
+}))
+
+const block = {
+  __typename: 'ButtonBlock',
+  id: 'block1.id',
+  children: []
+} as unknown as TreeBlock<ButtonBlock>
+
+const step1: TreeBlock<StepBlock> = {
+  id: 'step1.id',
+  __typename: 'StepBlock',
+  parentBlockId: null,
+  parentOrder: 0,
+  locked: false,
+  nextBlockId: null,
+  children: [block]
+}
+
+const stepBlockNextBlockUpdateMock: MockedResponse<
+  StepBlockNextBlockUpdate,
+  StepBlockNextBlockUpdateVariables
+> = {
+  request: {
+    query: STEP_BLOCK_NEXT_BLOCK_UPDATE,
+    variables: {
+      id: 'step0.id',
+      journeyId: 'journey-id',
+      input: {
+        nextBlockId: 'newStep.id'
+      }
+    }
+  },
+  result: {
+    data: {
+      stepBlockUpdate: {
+        __typename: 'StepBlock',
+        id: 'step0.id',
+        nextBlockId: 'newStep.id'
+      }
+    }
+  }
+}
+
+describe('useCreateStep', () => {
+  it('should create a new step block', async () => {
+    const cache = new InMemoryCache().restore({
+      'Journey:journey.id': {
+        blocks: [{ __ref: `StepBlock:step.id` }],
+        id: 'journey.id',
+        __typename: 'Journey'
+      },
+      'StepBlock:step1.id': {
+        ...step
+      }
+    })
+
+    mockUuidv4.mockReturnValueOnce('newStep.id')
+    mockUuidv4.mockReturnValueOnce('newCard.id')
+
+    const result = jest.fn().mockReturnValue(stepAndCardBlockCreateMock.result)
+
+    const { result: hook } = renderHook(() => useCreateStep(), {
+      wrapper: ({ children }) => (
+        <MockedProvider
+          mocks={[{ ...stepAndCardBlockCreateMock, result }]}
+          cache={cache}
+        >
+          <JourneyProvider value={{ journey: defaultJourney }}>
+            {children}
+          </JourneyProvider>
+        </MockedProvider>
+      )
+    })
+
+    await act(
+      async () =>
+        await hook.current({
+          x: -200,
+          y: 38
+        })
+    )
+
+    expect(result).toHaveBeenCalled()
+
```

Comment:
Can you add the cache check for blocks
```suggestion
expect(cache.extract()['Journey:journey-id']).toEqual({
    // step1 ref
    // newStep ref
    //newCard ref
})
```